### PR TITLE
🐛 Fixed missing grouping in not conditions and added list conversion for sigma expansion conditions

### PR DESCRIPTION
**Pull request by marcelkwaschny**

### CODE REVIEW

File: tests/test_backend_netwitness.py [hunk 1]
```diff
@@ -224,14 +224,17 @@ def test_netwitness_contains_modifier_with_list(netwitness_backend: NetWitnessBa
             detection:
                 sel:
                     fieldA|contains:
+                        # This is a comment
```
Comment (nikstuckenbrock) [inline tests/test_backend_netwitness.py:227]: Consider splitting this into multiple tests:
- `test_netwitness_contains_with_inline_comments`
- `test_netwitness_contains_with_single_value`
- `test_netwitness_contains_with_list`

File: sigma/backends/netwitness/netwitness.py [hunk 2]
```diff
@@ -147,15 +148,15 @@ class NetWitnessBackend(TextQueryBackend):
...
-    field_exists_expression: ClassVar[Optional[str]] = (
-        "exists({field})"  # Expression for field existence as format string with {field} placeholder for field name
-    )
+    # Expression for field existence as format string with {field} placeholder for field name
+    field_exists_expression: ClassVar[Optional[str]] = "{field} exists"
+
     # Expression for field non-existence as format string with {field} placeholder for field name.
     # If not set, field_exists_expression is negated with boolean NOT.
-    field_not_exists_expression: ClassVar[Optional[str]] = "notexists({field})"
+    field_not_exists_expression: ClassVar[Optional[str]] = "{field} !exists"
```
Comment (nikstuckenbrock) [inline sigma/backends/netwitness/netwitness.py:159]: Consider adding a const/variable for the negotiation character and dynamically build the strings containing it.

File: sigma/backends/netwitness/netwitness.py [hunk 3]
```diff
@@ -191,6 +192,67 @@ class NetWitnessBackend(TextQueryBackend):
...
+                return self.not_token + self.token_separator + self.convert_condition_group(arg, state)  # type: ignore
+
+            expr = self.convert_condition(arg, state)  # type: ignore
+            if isinstance(expr, DeferredQueryExpression):  # negate deferred expression and pass it to parent
+                return expr.negate()
+
+            # convert negated expression to string
+            return self.not_token + self.token_separator + self.group_expression.format(expr=expr)
+        except TypeError as error:
+            raise NotImplementedError("Operator 'not' not supported by the backend") from error
```
Comment (nikstuckenbrock) [inline sigma/backends/netwitness/netwitness.py:225]:
```suggestion
            raise NotImplementedError("Operator 'not' is not supported by the backend") from error
```

**Review by nikstuckenbrock — CHANGES_REQUESTED** [review]

File: sigma/backends/netwitness/netwitness.py (hunk 2 above)
Comment (marcelkwaschny) [reply sigma/backends/netwitness/netwitness.py:159]: The char is only used in this line therefore no need to replace it in multiple strings

File: tests/test_backend_netwitness.py [hunk 4]
```diff
@@ -461,6 +464,28 @@ def test_netwitness_less_equal_modifier(netwitness_backend: NetWitnessBackend):
...
+                product: test_product
+            detection:
+                selection:
+                    fieldname|windash|contains: -f
+                condition: selection
+            """
+        )
+    )
+
+    assert conversion_result == ["fieldname contains '-f','/f','–f','—f','―f'"]
```
[the remark this answers is not in the document]
Comment (marcelkwaschny) [reply tests/test_backend_netwitness.py:483]: Unfortunately there is no list or something else in the base sigma backend that I can use for this

File: tests/test_backend_netwitness.py (hunk 1 above)
Comment (marcelkwaschny) [reply tests/test_backend_netwitness.py:227]: I would like to keep it in a single file if that's okay for you :-)

**Review by nikstuckenbrock — APPROVED** [review]

**Pull request merged**